src/GoogleFit/FitOnFhir.GoogleFit.Tests/GoogleFitDataImporterTests.cs

```diff
@@ -72,22 +75,29 @@ public GoogleFitDataImporterTests()
         }
 
         [Fact]
-        public async Task GivenRefreshTokenReturnsDefault_WhenImportIsCalled_ImportReturns()
+        public async Task GivenRefreshTokenThrowsTokenRefreshException_WhenImportIsCalled_ImportReturns()
         {
-            AuthTokensResponse defaultAuthTokensResponse = default;
+            _usersTableRepository.GetById(
+                Arg.Is<string>(userid => userid == _userId),
+                Arg.Is<CancellationToken>(token => token == _cancellationToken)).Returns(_user);
 
+            // override ProcessDatasetRequests to throw an exception
+            string exceptionMessage = "token refresh exception";
+            var exception = new TokenRefreshException(exceptionMessage);
             _googleFitTokensService.RefreshToken(
                 Arg.Is<string>(userid => userid == _googleUserId),
-                Arg.Is<CancellationToken>(token => token == _cancellationToken)).Returns(defaultAuthTokensResponse);
+                Arg.Is<CancellationToken>(token => token == _cancellationToken)).Throws(exception);
 
             await _googleFitDataImporter.Import(_userId, _googleUserId, _cancellationToken);
 
-            await _googleFitClient.DidNotReceive().DataSourcesListRequest(
-                Arg.Is<string>(access => access == _accessToken),
+            await _usersTableRepository.Received(1).GetById(
+                Arg.Is<string>(usr => usr == _userId),
                 Arg.Is<CancellationToken>(cancel => cancel == _cancellationToken));
 
-            await _usersTableRepository.DidNotReceive().GetById(
-                Arg.Is<string>(usr => usr == _userId),
+            Assert.Equal(DataImportState.Unauthorized, _user.GetPlatformUserInfo().First().ImportState);
```

Comment:
I don't think this is a valid test. You actually need to make sure that the _usersTableRepository received a request to save the user object with the updated values.